config/v1/types_infrastructure.go

```diff
@@ -157,12 +209,45 @@ type PlatformStatus struct {
 	IBMCloud *IBMCloudPlatformStatus `json:"ibmcloud,omitempty"`
 }
 
+// AWSServiceEndpoint store the configuration of a custom url to
+// override existing defaults of AWS Services.
+type AWSServiceEndpoint struct {
+	// Name is the name of the AWS service.
```

Comment:
we need to restrict these names.  enumerated keys would be best.  Name restrictions would be second best.

Reply:
The names cannot be restricted. The number of AWS services are unbounded, and we can't play catchup with AWS SDK here. The requirement is users should be able to set endpoints for any AWS service.

Reply:
We need to at least document a naming convention for AWS services. We can't have customers define `S3` as the service name, and have things like the image registry not pick up the endpoint because its operator is expecting `s3` as the service name...

Reply:
Here's an idea - for a given AWS service:

1. Drop any "AWS" or "Amazon" prefix.
2. Drop any "Service" suffix.
3. If the service has a well-established acronym in AWS documentation, use that in all capitals. Example: `EC2`, `S3`, `ELB`, `VPC`, `RDS`.
4. Otherwise, use the service name in `CamelCase`. Ex: `Route53`, `ElastiCache`.

Reply:
@adambkaplan please no, AWS services already have names in their SDK:

https://github.com/aws/aws-sdk-go/blob/fa81b5cb4456a1e77f5ccd74d729d1761821aea7/models/endpoints/endpoints.json#L68

Reply:
It makes sense to reference. A quick scan seems to reveal that they don't support emoji characters.  Anything other than a-z,-,0-9?  We can start restrictive, but catching obvious typos like that seems like a good idea.